src/services/shop/shop-router.ts

```diff
@@ -42,148 +43,423 @@ shopRouter.get(
+            }
+
+            // Clear the user's order from the cart
+            await Models.ShopOrder.deleteOne({ userId });
+            return res.status(StatusCode.SuccessOK).json(order);
+        } catch (error) {
+            console.error("Error processing order:", error);
+            return res.status(StatusCode.ServerErrorInternal).json(ShopInternalError);
+        }
     },
 );
 
-shopRouter.put(
-    "/item/:id/",
+shopRouter.post(
+    "/cart/:itemId",
     specification({
-        method: "put",
-        path: "/shop/item/{id}/",
+        method: "post",
+        path: "/shop/cart/{itemId}/",
         tag: Tag.SHOP,
-        role: Role.ADMIN,
-        summary: "Updates a shop item",
+        role: Role.USER,
+        summary: "Adds item to users cart",
         parameters: z.object({
-            id: ShopItemIdSchema,
+            itemId: ShopItemIdSchema,
         }),
-        body: ShopItemUpdateRequestSchema,
         responses: {
             [StatusCode.SuccessOK]: {
-                description: "The new item",
-                schema: ShopItemSchema,
+                description: "The successfully updated order",
+                schema: ShopOrderInfoSchema,
             },
             [StatusCode.ClientErrorNotFound]: {
-                description: "Item doesn't exist",
+                description: "Shop Item DNE",
                 schema: ShopItemNotFoundErrorSchema,
             },
+            [StatusCode.ClientErrorBadRequest]: {
+                description: "Not enough quantity in shop",
+                schema: ShopInsufficientQuantityErrorSchema,
+            },
+            [StatusCode.ClientErrorPaymentRequired]: {
+                description: "Not enough points to purchase",
+                schema: ShopInsufficientFundsErrorSchema,
+            },
+            [StatusCode.ServerErrorInternal]: {
+                description: "Errors that should never happen",
+                schema: ShopInternalErrorSchema,
+            },
         },
     }),
     async (req, res) => {
-        const { id: itemId } = req.params;
-        const updateRequest = req.body;
+        const { itemId } = req.params;
+        const { id: userId } = getAuthenticatedUser(req);
+
+        let userOrder = await Models.ShopOrder.findOne({ userId: userId });
+        //user doesn't have a order yet
+        if (!userOrder) {
+            const shopOrder: ShopOrder = {
+                items: [],
+                quantity: [],
+                userId: userId,
+            };
+
+            userOrder = await Models.ShopOrder.create(shopOrder);
+        }
 
-        const updatedItem = await Models.ShopItem.findOneAndUpdate({ itemId }, updateRequest, {
-            new: true,
-        });
+        // This should never get hit, just checking here so typescript doesn't get mad
+        if (!userOrder) {
+            return res.status(StatusCode.ServerErrorInternal).send(ShopInternalError);
+        }
```

Comment:
This is a good example of where you don't need a custom error - if userOrder can be null, just throw an error like "User order missing". Since this shouldn't happen normally, it's fine.